include/libhal-icm/icm20948.hpp

```diff
@@ -24,121 +24,118 @@ class icm20948
 {
 
 public:
-
-typedef enum ICM20948_CYCLE
-  {
-    ICM20948_NO_CYCLE = 0x00,
-    ICM20948_GYR_CYCLE = 0x10,
-    ICM20948_ACC_CYCLE = 0x20,
-    ICM20948_ACC_GYR_CYCLE = 0x30,
-    ICM20948_ACC_GYR_I2C_MST_CYCLE = 0x70
-  } ICM20948_cycle;
-
-  typedef enum ICM20948_INT_PIN_POL
+  enum class icm20948_cycle : hal::byte
   {
-    ICM20948_ACT_HIGH,
-    ICM20948_ACT_LOW
-  } ICM20948_intPinPol;
+    icm20948_no_cycle = 0x00,
```

Comment:
Fully expressed from outside the scope of this library to acquire this enum's value you have to type the following:

```suggestion
    auto no_cycle = hal::icm::icm20948::icm20948_cycle::icm20948_no_cycle;
```

Thats a lot of `icm20948`s being typed. We can reduce the name to just `cycle` for the enum and remove the prefixes in front of each enum value. In C you have to do this because enums propagate to the scope they are defined in, meaning that name collisions are very possible. This means you need to prefix everything. But in C++ you get the namespace scoping for you, so need to add more of it.

A better alternative would be:

```C++
auto no_cycle = hal::icm::icm20948::cycle::no_cycle;
```

The first `hal::icm` is still required and so is the `icm20948` to prevent confusion if a user is using `using namespace hal::icm` for some reason. The other items can be trimmed.